### PR TITLE
Weekly timetable view

### LINKED ISSUE #7: Show Multiple Days for Timetable
It may be useful to see all events in the upcoming few days instead of just for the current day. A 3-day view might work best in portrait mode.

### PR DESCRIPTION
**Pull request by hueyy**
Resolves #7, makes #426 obsolete

[https://expo.io/@uclapi/ucl-assistant?release-channel=dev.feature-weekly_timetable](https://expo.io/@uclapi/ucl-assistant?release-channel=dev.feature-weekly_timetable)

### CODE REVIEW

**Comment by hueyy** [comment]
I'm hoping to be able to put the week number at the top at some point as well. I never know what week it is, so will actually be very useful for me.

File: selectors/timetableSelectors.js [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+  )
+}
+
+export const weeklyTimetableArraySelector = createSelector(
+  [timetableSelector],
+  (timetable) => [
+    null,
+    ...padTimetable(
+      _.chunk(
+        Object.entries(timetable)
```
Comment (TuxedoFish) [inline selectors/timetableSelectors.js]: I get a "Cannot convert undefined or null to object" error when I try to run this locally

Comment (TuxedoFish) [reply selectors/timetableSelectors.js]: Added 
```
(timetable) => {
    if(typeof timetable==="undefined") {
      return []
    } else {
``` 
and it seems to work fine looks awesome

**Comment by hueyy** [comment]
I considered putting the module names instead of the module code, but I realised some modules have very long names, e.g. Writing Composers: The Fiction of Music in the Long Nineteenth Century, which overflows onto several lines, and doesn't look great. One possible solution could be to show the module name if it's short (maybe less than 10 chars?), otherwise show the module code, but that could be confusing.

New data is fetched (and the loading indicator appears) if the data has never been fetched, is stale, or the view is the one for today (where it assumes you're trying to refresh). I could remove the refresh for the 3rd scenario if you think it makes sense. 

I'll adjust the top & bottom padding on the nothing scheduled screen, and center align the date and last updated text.

**Comment by hueyy** [comment]
Oh yes I forgot I created a new attribute in the redux store and new clients will have it undefined. Nice find, thanks.

**Comment by hueyy** [comment]
Made the suggested changes:

**Comment by TuxedoFish** [comment]
> I considered putting the module names instead of the module code, but I realised some modules have very long names, e.g. Writing Composers: The Fiction of Music in the Long Nineteenth Century, which overflows onto several lines, and doesn't look great. One possible solution could be to show the module name if it's short (maybe less than 10 chars?), otherwise show the module code, but that could be confusing.
> 
> New data is fetched (and the loading indicator appears) if the data has never been fetched, is stale, or the view is the one for today (where it assumes you're trying to refresh). I could remove the refresh for the 3rd scenario if you think it makes sense.
> 
> I'll adjust the top & bottom padding on the nothing scheduled screen, and center align the date and last updated text.

Yeah thats's a good point, I was trying to think how google does it but actually they just have a small font for the title and it looks nicer on this with the green module code! I agree keep it how it is!

I realised actually I had been pressing the arrows not swiping so what I meant was that it doesnt say loading when you click the buttons but I only notice that because it takes a long time to swap whereas the swiping is almost instant! I don't know if you get this too or if this is just an expo being slow thing? But on that note the swiping works super well.

Looks good!! Oops I meant the nothing scheduled on a particular day like this:

Screenshot_20200101-144634_Expo

It was a really picky thing but like it looks like the "nothing scheduled" doesn't appear vertically aligned. I actually really like the nothing scheduled for the week screen!

**Comment by hueyy** [comment]
Ugh yeah I noticed the delay when navigating via the buttons. It's slightly better in production mode than development mode, but it seems to be a recurring issue with one of the components I'm using (https://github.com/leecade/react-native-swiper/issues/947). 

Changed the nothing scheduled text as suggested.

**Comment by TuxedoFish** [comment]
@hueyy oh right thats annoying :( Happy to squash and merge then!

**Comment by Feverfew** [comment]
@hueyy didn't realise this would get reviewed quickly... Would seem preferable to have the day above than on the side? horizontal space is minimal but vertical space is "infinite" (you can scroll?)

**Comment by hueyy** [comment]
I experimented with that, but I found it was visually easier to break up the week into days and identify which event was under which day with the current horizontal view rather than a vertical view.

**Pull request merged**